docs/notebooks/Poincare Evaluation.ipynb

```diff
@@ -0,0 +1,1553 @@
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-np-embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Apply patches\n",
+    "if not patches_applied:\n",
+    "    % cd {cpp_repo_name}\n",
+    "    ! git apply ../poincare_burn_in_eps.patch\n",
+    "\n",
+    "    % cd ../{np_repo_name}\n",
+    "    ! git apply ../poincare_numpy.patch\n",
+    "    \n",
+    "    patches_applied = True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 124,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding/work\n",
+      "-- Configuring done\n",
+      "-- Generating done\n",
+      "-- Build files have been written to: /home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding/work\n",
+      "[100%] Built target poincare_embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Compile the code for the external c++ implementation into a binary\n",
+    "% cd {parent_directory}/{cpp_repo_name}\n",
+    "! mkdir -p work\n",
+    "% cd work\n",
+    "! cmake ..\n",
+    "! make\n",
+    "% cd {current_directory}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You might need to install an updated version of `cmake` to be able to compile the source code. Please make sure that the binary `poincare-embedding` has been created before proceeding."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Training\n",
+    "\n",
+    "### 2.1 Create the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# These directories are auto created in the current directory for storing poincare datasets and models\n",
+    "data_directory = os.path.join(parent_directory, 'data')\n",
+    "models_directory = os.path.join(parent_directory, 'models')\n",
+    "\n",
+    "# Create directories\n",
+    "! mkdir -p {data_directory}\n",
+    "! mkdir -p {models_directory}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "82115 nouns\n",
+      "743241 hypernyms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Prepare the WordNet data\n",
+    "wordnet_file = os.path.join(data_directory, 'wordnet_noun_hypernyms.tsv')\n",
+    "! python {parent_directory}/poincare-embedding/scripts/create_wordnet_noun_hierarchy.py {wordnet_file}"
```

Comment:
After correcting the path I'm getting:
```python
Traceback (most recent call last):
  File "/data_ssd/poincare/gensim/docs/notebooks/poincare/poincare-cpp-embedding/scripts/create_wordnet_noun_hierarchy.py", line 4, in <module>
    from nltk.corpus import wordnet as wn
ImportError: No module named nltk.corpus
```

We may need to add `nltk` to requirements.

Reply:
... and `click` as well.

Reply:
Thinking more about it, it doesn't seem a great idea to make these gensim requirements just because of the notebook. Let's just do
```
import nltk
import click
```
in the first cell in the notebook so that it fails there if these libs are missing. It should be obvious for the user that they need to install these. (I don't expect non-expert users to ever run the notebook.)

Reply:
Yeah, makes sense. I added the imports at the beginning, and a link to a separate `requirements.txt` in `poincare/` for easy installation of missing imports.